modules/ocf_desktop/files/stats/update.sh

```diff
@@ -1,15 +1,21 @@
 #!/bin/bash
-# imvirt spews dmesg errors on stderr
-# https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=842226
-[ "$(imvirt 2>/dev/null)" == "Physical" ] || exit 0 # only count hours on desktops
+
+# state can be "active" "inactive" "cleanup"
+# "active" implies a current logged-in session
+# "inactive" implies the desktop isn't asleep but no one's logged in (noop)
+# "cleanup" is a semi-hack to eagerly clean up sessions without relying
+#   on the periodic session cleanup, or making too many calls to ocfweb
+#   (and by extension, mysql)
+# other files to look at: session-cleanup and session-setup in /etc/lightdm
 
 CUR_USER=$(who | awk '$NF == "(:0)" { print $1 }')
-DATA="state=inactive"
+STATE="inactive"
 
-if [ -n "$CUR_USER" ]; then
-	DATA="state=active&user=$CUR_USER"
-fi
+[[ -n "$CUR_USER" ]] && STATE="active" || STATE=${1:-"inactive"}
 
-curl --data "$DATA" \
-    https://labstats.ocf.berkeley.edu:444/update.cgi \
-	2>/dev/null
+# use 3 states for this minor optimization: only make requests to ocfweb
+# when a user is logged or once during cleanup, otherwise, stay silent
+if [ "$STATE" != "inactive" ]; then
```

Comment:
What's the point of having an inactive state if it's never used? This was why I was confused yesterday, because it seems to me you only need two states, one for when a user is logged in (`active`), and another one that fires when a user logs out (`inactive` or `cleanup`, whichever you prefer). Then, as long as the logout script fires (or even if it doesn't, due to the `_cleanup_sessions()` function), then the user should be logged out as long as you send whatever this end state is. Essentially you only need to update if a user is logged in, or on cleanup, which I think only needs 2 states overall.

Reply:
Ah, I see what you mean. I've been treating `inactive` as a flag but it can be implied by
```
[[ -n "$CUR_USER" ]] && STATE="active" || STATE="$1"
if [[ -n "$STATE" ]]; then
  curl ...
fi
```
while maintaining the same optimization, and only using `active` and `cleanup` as explicit states.